### PR TITLE
maintenance: cleanup of xc and dependencies

### PR DESCRIPTION
**Pull request by psafont**
xc had some code that was not compiled:
- list_domains was in another directory, it's been deleted
- xenstore_readdir was using ezxenstore functionality that has been
  commented out, making it uncompilable; it's been deleted
- cancel_utils_test was using some nonexisting functions and obsoleted
  interfaces that have been updated

dependencies got refreshed to more accurately replect the actual
dependencies and rules got added in xc to build some binaries that were
not buildable, as well as some tests that only run with xenstore present

### CODE REVIEW

File: xc/cancel_utils_test.ml [hunk 1]
```diff
@@ -28,50 +28,27 @@ let xenstore_test xs _ =
...
-  let task = Xenops_task.add tasks "test" (fun _ -> None) in
-  let (_ : Thread.t) =
-    Thread.create
-      (fun () ->
-        Thread.delay 1. ;
-        Xenops_task.cancel tasks task.Xenops_task.id)
-      ()
-  in
-  try
-    let _, _ = cancellable_subprocess task "/bin/sleep" ["3s"] in
```
Comment (psafont) [inline xc/cancel_utils_test.ml:52]: This function was not available anymore

**Comment by psafont** [comment]
I've left this as a draft because linking of the xenopsd-xc binary fails because of the xenctrl bindings don't seeme to be available. I haven't had time to fix this issue.

**Comment by lindig** [comment]
As part of the cleanup you could also clean up directory permissions, which are a bit too wide for a good default.

**Comment by psafont** [comment]
> As part of the cleanup you could also clean up directory permissions, which are a bit too wide for a good default.

I don't think git can do this, it only seems to track whether a file is executable (755) or not (644): https://git-scm.com/docs/git-update-index#Documentation/git-update-index.txt---chmod-x

File: xc/cancel_utils_test.ml [hunk 2]
```diff
@@ -28,50 +26,27 @@ let xenstore_test xs _ =
...
-    "cancel test"
-    >::: ["subprocess" >:: subprocess_test; "xenstore" >:: xenstore_test xs]
-  in
-  run_test_tt ~verbose:!verbose suite
+  try
+    Xenstore.with_xs (fun xs ->
+        let suite = "cancel test" >::: ["xenstore" >:: xenstore_test xs] in
+        run_test_tt ~verbose:!verbose suite |> ignore)
+  with Xs_transport.Could_not_find_xenstore ->
+    (* ignore test, we're not running on domain 0 *)
```
Comment (lippirk) [inline xc/cancel_utils_test.ml:51]: the behaviour has changed here? previously it looks like we ran `exit 1`

Comment (psafont) [reply xc/cancel_utils_test.ml:51]: That's correct, I don't want the test to fail because it's not running on dom0 as now it's run on every CI job.

There is still text being output when this connection fails, so it's not needed to add this here. This is only useful when testing this manually, I don't think the test was ever run in an automated way.

```
Failed to find xenstore socket. I tried the following:
  /var/run/xenstored/socket
  /proc/xen/xenbus
  /dev/xen/xenstore

On linux you might not have xenfs mounted:
   sudo mount -t xenfs xenfs /proc/xen
Or perhaps you just need to set the XENSTORED_PATH environment variable.
```

Comment (lippirk) [reply xc/cancel_utils_test.ml:51]: Fair enough. I suppose we could use environment variables to detect whether we are running it in an automated way, but that might be too brittle

File: xc/cancel_utils_test.ml [hunk 3]
```diff
@@ -28,50 +26,27 @@ let xenstore_test xs _ =
     Thread.create
       (fun () ->
         Thread.delay 1. ;
-        Xenops_task.cancel tasks task.Xenops_task.id)
+        Xenops_task.with_cancel task (fun () -> ()) (fun () -> ()))
```
Comment (lippirk) [inline xc/cancel_utils_test.ml:29]: `Fun.id`?

Comment (psafont) [reply xc/cancel_utils_test.ml:29]: The current function is more restrictive, type-wise. I like it better.
